tests/components/homekit/test_type_cameras.py

```diff
@@ -122,28 +174,46 @@ async def test_camera_stream_source_configured(hass, run_driver, events):
         stdout_pipe=False,
     )
 
+    await _async_setup_endpoints(hass, acc)
+    working_ffmpeg = _get_working_mock_ffmpeg()
+    session_info = acc.sessions[MOCK_START_STREAM_SESSION_UUID]
+
     with patch(
         "homeassistant.components.demo.camera.DemoCamera.stream_source",
         return_value="rtsp://example.local",
     ), patch(
         "homeassistant.components.homekit.type_cameras.HAFFmpeg",
         return_value=working_ffmpeg,
     ):
-        acc.set_selected_stream_configuration(MOCK_START_STREAM_TLV)
-        await acc.stop_stream(session_info)
+        await _async_start_streaming(hass, acc)
+        await _async_stop_all_streams(hass, acc)
         # Calling a second time should not throw
-        await acc.stop_stream(session_info)
-        await hass.async_block_till_done()
+        await _async_stop_all_streams(hass, acc)
 
-    assert await hass.async_add_executor_job(acc.get_snapshot, 1024)
+    turbo_jpeg = mock_turbo_jpeg(
+        first_width=16, first_height=12, second_width=300, second_height=200
+    )
+    with patch(
+        "homeassistant.components.homekit.img_util.TurboJPEG", return_value=turbo_jpeg
+    ):
+        TurboJPEGSingleton()
+        assert await hass.async_add_executor_job(
+            acc.get_snapshot, {"aid": 2, "image-width": 300, "image-height": 200}
+        )
+        # Verify the bridge only forwards get_snapshot for
+        # cameras and valid accessory ids
+        assert await hass.async_add_executor_job(
+            bridge.get_snapshot, {"aid": 2, "image-width": 300, "image-height": 200}
+        )
 
-    # Verify the bridge only forwards get_snapshot for
-    # cameras and valid accessory ids
-    assert await hass.async_add_executor_job(bridge.get_snapshot, {"aid": 2})
     with pytest.raises(ValueError):
-        assert await hass.async_add_executor_job(bridge.get_snapshot, {"aid": 3})
+        assert await hass.async_add_executor_job(
+            bridge.get_snapshot, {"aid": 3, "image-width": 300, "image-height": 200}
+        )
     with pytest.raises(ValueError):
-        assert await hass.async_add_executor_job(bridge.get_snapshot, {"aid": 4})
+        assert await hass.async_add_executor_job(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)